### ISSUE TITLE
Garage door control via Internet for Genie controller

### ISSUE THREAD

**Issue by 130s** [kinu-garage/hut_10sqft#560]
Same as https://github.com/130s/30y-130s_life/issues/46 (for LifeMaster), I want to be able to operate our Genie garage door controller via the Internet.

Can be done via `ring.com` ideally, but if not I can rely on home automation framework.

**Comment by 130s** [comment]
Multiple compatibility checks are needed for Internet-based garage opener (in my case, only "C) garage door controller" is locked in to `Genie` already. Others can be decided based on the result):
- A) your home automation system(s) <--> B) your garage door Internet-based remote
   - [ ] ring.com
   - [ ] Home Assistant
   - [ ] openHAB
- [ ] B) your garage door Internet-based remote <--> C) garage door controller

If the one ring.com supports, [Chamberlain® Smart Garage® Control (ring.com)](https://ring.com/products/chamberlain-smart-garage-control-door-opener), is compatible with our Genie model (B-C link above), then I'll go for it, as link A-B is certified by ring.

In the previous case https://github.com/130s/30y-130s_life/issues/46#issuecomment-244519190 B was `GD00Z-4`, which seems to support some Genie devices (but I need to find the model of ours) and C was LiftMaster.

**Comment by 130s** [comment]
> -  B) your garage door Internet-based remote <--> C) garage door controller

- [This page on support.chamberlaingroup.com](https://support.chamberlaingroup.com/s/article/List-of-compatible-garage-door-openers-for-MyQ-Garage-1484145635905) shows a bunch of Genie products. So I'm hopeful. Still need to find our model(s).

**Comment by 130s** [comment]
>>  I need to find the model of ours)
>
>  This page on support.chamberlaingroup.com shows a bunch of Genie products. So I'm hopeful. Still need to find our model(s).

Yup, we need to know the model of Genie device.

**Comment by 130s** [comment]
> A) your home automation system(s) <--> B) your garage door Internet-based remote

Still unclear.

https://www.twice.com/product/product-spotlight/smart-home/the-genie-company-announces-garage-door-integration-for-brilliant-smart-home-system says the following, but I don't want to add yet another framework unless I absolutely need to. 
> Brilliant currently works with Amazon Alexa, Google Assistant, HomeKit, Sonos, Ring, Schlage, Honeywell, Ecobee, Philips Hue, LIFX, Wemo, ButterflyMX, and more, with new integrations added on a monthly basis. Brilliant also supports hubs such as Samsung SmartThings for connecting Z-Wave or Zigbee products and can connect directly to bridge products from Philips Hue and others.

Worth asking on ring.com community.

**Comment by 130s** [comment]
>> A) your home automation system(s) <--> B) your garage door Internet-based remote

- If "A" == ring.com:
   - https://www.myq.com/works-with-myq lists ring.com. We should stick to it if we choose to hook garage opener to ring.com.
      - `MyQ` is manufacturer-independent architecture IINM that connects garage opener and wireless technology, particularly Wifip.
      - Back in 2016 https://github.com/130s/30y-130s_life/issues/92#issuecomment-183541488 I disliked MyQ for:
         - its overkilling technology usage (Wifi for smarthome isn't smart when more suited technologies e.g. z-wave are available)
	 - its dedicated, unintegratable GUI requirement. But maybe ring.com integrates UI over the years?
- If "A" == something else:
   - Genie made its own wifi connect architechture "Aladdin Connect".
      - There's `Home Assistant` binding to it https://www.home-assistant.io/integrations/aladdin_connect/
      - Google Home is listed up https://www.geniecompany.com/aladdin-connect-smart-home-partners
   - Someone said `Garadget` https://community.home-assistant.io/t/best-garage-door-controller/9801/2 but I see less usecases so I worry about its maintenance status.

**Comment by 130s** [comment]
TLDR; Going with Z-Wave, GoControl GD00Z-8.

# Choice for garage controller remote hook
Leaning toward either `Aladdin Connect` or [MyQ](https://www.home-assistant.io/integrations/myq/).
- Aladdin Connect`: Very few users according to https://www.home-assistant.io/integrations/aladdin_connect/ (210 installations as of today. Not even 1%). There's an issue recently found and fixed https://github.com/home-assistant/core/issues/65665. It's good many users were paying attention in short period of time.
- `MyQ`: 4.4% today, which is beyond 4K installations?
   - I disliked MyQ in 2016 as I mentioned https://github.com/130s/hut_10sqft/issues/560#issuecomment-933192465, but time passed and 1) now I'm liking whatever that has broader user base, which `MyQ` seems to qualify esp. with their likely-successful manufacturer-agnostic approach, 2) HoA provides integrated UI so MyQ providing its own UI shouldn't matter to me.
   - Is MyQ compatible with our Genie garage controller?
      - The garage guy https://github.com/130s/30y-130s_life/issues/1174 said they'd come and install MyQ if we like.
      - https://www.myq.com/app/myq-compatibility shows yes for Genie -> ScrewDrive series IC.
         > You have a myQ-compatible garage door opener.* All you need is the myQ Smart Garage Control to start enjoying all that myQ has to offer.
- Z-Wave options (same as when I DIYed with LiftMaster https://github.com/130s/30y-130s_life/issues/46)
   - [GoControl GD00Z-8 (amazon.com)](https://www.amazon.com/GoControl-GD00Z-8-GC-Z-Wave-Security-Black/dp/B085LKPHK6/ref=sr_1_3?keywords=zwave+garage+door+opener&qid=1654150257&sr=8-3)
      - Is this an upgrade product from [GD00Z-4, the one I used 7 years ago (blogspot.com)](http://gm130s.blogspot.com/2016/12/garage-door-control-over-internet-using.html)?
   - I haven't found HoA entity specific for `GD00Z-` though, some ppl said they were able to show the monitoring and controll I/F on Lovelace, e.g. [community.home-assistant.io#304229](https://community.home-assistant.io/t/z-wave-garage-door-opener-with-zwavejs2mqtt/304229/14).

# What does garage controller need in order to connect to the remote hook

- Aladdin
   - So-called "`Aladdin Connect Retrofit Kit`" seems to be the one that connects to a garage controller and wifi. So this should be what I'm looking for.
   - Genie product model ID: CM7600IC/A (according to [the pics](https://drive.google.com/file/d/1RKsOCnY0Yaz9sKOuuPSGi-NE5Rt2ieV3/view?usp=sharing)). AFAIURC, it's released in 1995 https://github.com/130s/30y-130s_life/issues/1017#issuecomment-1144459064 so it should be compatible with Retrofit Kit according to [compatibility doc (geniecompany.com)](https://www.geniecompany.com/_files/ugd/3ef5a4_0f433257d2fd4f55a64d2875e61bd3c3.pdf)?
- MyQ: TBD
- `GD00Z-8`: I assume it's similar to what I did the [last time with GD00Z-4 + 2gig + alarm.com](http://gm130s.blogspot.com/2016/12/garage-door-control-over-internet-using.html).

**Comment by 130s** [comment]
Related but off topic: I found 2 pairs of terminals   instead of 1,each of which consists of blue and white-blue stripe cable.
![PXL_20220609_145946841.jpg](https://user-images.githubusercontent.com/1840401/172880020-768bb933-7fe5-4ee0-a458-0c264f846de1.jpg)

Feel like figuring out what these 2 pairs are for (if these are connected to push button, I'm aware of only 1 button connected to this controller so why 2 pairs?).

- Unwired the pair on right side
   - Pressing wall button triggers closing the garage, but once I release the button the garage started moving back.
   - Keeping button pressed let the garage closed til the end. But once I released the button, it went back all the way again.
- Unwired both pairs. -> Wall button doesn't seem to trigger anything.
- Unwired the left pair. -> Wall button doesn't seem to trigger anything.

I concluded terminals 1 and 2 are for wall button (3 - and4 are for something called safe-T, which I'm yet to figure out).

**Comment by 130s** [comment]
Done https://github.com/130s/30y-130s_life/issues/1021#issuecomment-1160823468

**Comment by 130s** [comment]
Just for the record, I think I forgot to write why I went with Z-Wave, DIY options while back then I was leaning toward proprietary options. I guess initially I was interested in hooking the garage monitoring to Ring.com so that was a limiting factor hence proprietary options, but over months I found Home Assistant was much more capable than I thought so I switched my policy to minimize the reliance on ring.com and rely on HASS for everything in the household automation stuff.
